wireapp/wire-android-reloaded
# fix: too frequent conversation list updates

----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues

Conversation list updates were being done too frequently, especially when first launching the app and because of that the whole list was being rearranged every time.

### Causes (Optional)

When fixing one issue with 1-1 conversations with deleted user, we created this one, because until now empty item was emitted at the beginning for each conversation and because of that every actual update with proper data (for every conversation independently) was emitting the whole list again.

### Solutions

Updated kalium reference and handled passing storage failure through the flow (we will need to handle failures properly on screens).

Needs releases with:

- https://github.com/wireapp/kalium/pull/726

### Testing

#### Test Coverage (Optional)


#### How to Test

Open the app and see if the list of conversations flickers (changes the order and size multiple times).

### Attachments (Optional)

#### With flickering:

https://user-images.githubusercontent.com/30429749/181587582-7bd49432-d2df-46ec-841a-d6a4c34463c7.mp4

&nbsp;
#### Without flickering:

https://user-images.githubusercontent.com/30429749/181587730-86a78809-ea17-48e3-bc70-e66769e0e1f3.mp4

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
